src/terrain/chunk.hpp

```diff
@@ -45,6 +45,7 @@ class Terrain;
 class Chunk : public voxel_utility::VoxelBase {
     std::list<NodeGroup> node_groups_;
     Terrain* ter_;
+    TerrainBase* terrain_base_;
```

Comment:
Why does terrain not inherit from this?

Reply:
I asked you about this and didn't get a response so this was the solution.

Basically chunk needs get_voxel to be inline (for speed). Therefore terrain base must be included in chunk.
Chunk needs to be able to accesses voxels without a function jump.

terrain base -> chunk -> terrain
terrain base -> terrain

I didn't know of a way to have terrain inherit from terrain base, and include the methods from terrain base in the chunk header.

Reply:
Responded in discord. Double inlining is a thing.